adapters/ing-adapter/src/test/java/de/adorsys/xs2a/adapter/ing/IngServiceProviderTest.java

```diff
@@ -0,0 +1,111 @@
+
+import de.adorsys.xs2a.adapter.api.AccountInformationService;
+import de.adorsys.xs2a.adapter.api.Oauth2Service;
+import de.adorsys.xs2a.adapter.api.PaymentInitiationService;
+import de.adorsys.xs2a.adapter.api.Pkcs12KeyStore;
+import de.adorsys.xs2a.adapter.api.http.HttpClientConfig;
+import de.adorsys.xs2a.adapter.api.http.HttpClientFactory;
+import de.adorsys.xs2a.adapter.api.model.Aspsp;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import java.io.ByteArrayInputStream;
+import java.io.InputStream;
+import java.nio.charset.StandardCharsets;
+import java.security.*;
+import java.security.cert.CertificateException;
+import java.security.cert.CertificateFactory;
+import java.security.cert.X509Certificate;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+class IngServiceProviderTest {
+
+    private IngServiceProvider serviceProvider;
+    private final HttpClientConfig httpClientConfig = mock(HttpClientConfig.class);
+    private final HttpClientFactory httpClientFactory = mock(HttpClientFactory.class);
+    private final Pkcs12KeyStore keyStore = mock(Pkcs12KeyStore.class);
+    private final X509Certificate certificate = generateCertificate();
+    private final PrivateKey privateKey = generatePrivateKey();
+    private final Aspsp aspsp = new Aspsp();
+
+    @BeforeEach
+    void setUp() throws KeyStoreException, UnrecoverableKeyException, NoSuchAlgorithmException {
+        serviceProvider = new IngServiceProvider();
+        when(httpClientFactory.getHttpClientConfig()).thenReturn(httpClientConfig);
+        when(httpClientConfig.getKeyStore()).thenReturn(keyStore);
+        when(keyStore.getQsealCertificate(any())).thenReturn(certificate);
+        when(keyStore.getQsealPrivateKey(any())).thenReturn(privateKey);
+    }
+
+    @Test
+    void getAccountInformationService() {
+        AccountInformationService actualService
+            = serviceProvider.getAccountInformationService(aspsp, httpClientFactory, null);
+
+        assertThat(actualService)
+            .isInstanceOf(IngAccountInformationService.class);
+    }
+
+    @Test
+    void getOauth2Service() {
+        Oauth2Service actualService
+            = serviceProvider.getOauth2Service(aspsp, httpClientFactory);
+
+        assertThat(actualService)
+            .isInstanceOf(IngAccountInformationService.class);
+    }
+
+    @Test
+    void getPaymentInitiationService() {
+        PaymentInitiationService actualService
+            = serviceProvider.getPaymentInitiationService(aspsp, httpClientFactory, null);
+
+        assertThat(actualService)
+            .isInstanceOf(IngPaymentInitiationService.class);
+    }
+
+    private static X509Certificate generateCertificate() {
+        X509Certificate certificate = null;
+        CertificateFactory cf = null;
+
+        String b64 =
```

Comment:
Maybe move this to a separate variable?

Reply:
you mean, make a class variable instead of a method one?